bundles/binding/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/TACmiMeasureType.java

```diff
@@ -35,18 +67,26 @@ public int getMeasure() {
 
     /**
      * Return the measure type for the specified name.
-     * 
+     *
      * @param measure
      * @return
      */
     public static TACmiMeasureType fromString(String measure) {
         if (!StringUtils.isEmpty(measure)) {
-            for (TACmiMeasureType measureType : TACmiMeasureType.values()) {
-                if (measureType.toString().equalsIgnoreCase(measure)) {
-                    return measureType;
-                }
-            }
+            return TACmiMeasureType.valueOf(measure.toUpperCase());
         }
         throw new IllegalArgumentException("Invalid measure: " + measure);
     }
+
+    /**
+     * Return measure type for a specific int value
+     */
+    public static TACmiMeasureType fromInt(int type) {
+        if (type >= 0 && type <= 21) {
+            return measureTypes[type];
+        } else {
+            logger.warn("Received unexpected measure type {}", type);
+            return TACmiMeasureType.UNSUPPORTED;
```

Comment:
Would it not be better to leave this enum ignorant of how it's used, but instead just throw an exception like is done in `fromString` above, for consistency and also keeping concerns separated?  Then the user of this enum will catch the exception and log from that layer.